modules/core/shared/src/main/scala/lucuma/core/math/dimensional/unit.scala

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.math.dimensional
+
+import coulomb.define._
+
+// All of this is a bridge between coulomb an runtime quantities (as defined in `Qty`).
+
+/**
+ * Runtime representation of a physical unit. Wraps:
+ *   - A type (usually phantom), meant to be `coulomb`'s type for the unit; and
+ *   - A `UnitDefinition`, which is a common trait of `coulomb`'s `BaseUnit` and `DerivedUnit`.
+ *
+ * In `coulomb` units are types. However in some cases we want to have runtime representations of
+ * units.
+ */
+trait UnitType { self =>
+  type Type
+  def definition: UnitDefinition
+
+  def withValue[N](_value: N): Qty[N] = new Qty[N] {
+    val value = _value
+    val unit  = self
+  }
+}
+
+/**
+ * Type-parametrized runtime representation of physical unit `U`.
+ *
+ * Can be automatically derived if there's an implicit `BaseUnit[U]` or `DerivedUnit[U, _]` in
+ * scope.
+ */
+trait UnitOfMeasure[U] extends UnitType {
```

Comment:
Is this something that should be in `coulomb`?

Reply:
If `coulomb` had a common ancestor of `BaseUnit[U]` and `DerivedUnit[U, D]` that was typed in `U`, we could use it and wouldn't need this.